velero-plugins/imagestream/shared.go

```diff
@@ -78,5 +102,64 @@ func findSpecTag(tags []imagev1API.TagReference, name string) *imagev1API.TagRef
 	return nil
 }
 
+func getRoute(namespace string, location string, configMap string) (string, error) {
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "Error in cluster config", err
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "Error in getting client", err
+	}
+	cMap := client.CoreV1().ConfigMaps(namespace)
+	mapClient, err := cMap.Get(configMap, metav1.GetOptions{})
+	if err != nil {
+		return "Error in getting config map", err
+	}
+	osClient, err := routev1client.NewForConfig(config)
+	if err != nil {
+		return "Error in getting client for route", err
+	}
+	routeClient := osClient.Routes(namespace)
+	route, err := routeClient.Get(mapClient.Data[location], metav1.GetOptions{})
+	if err != nil {
+		return "No route found", err
+	}
+	return route.Spec.Host, nil
+}
+
+func getBackup(name string, namespace string) (string, error) {
```

Comment:
NIT: `getBackupStorageLocationForBackup`. Something implying this is returning the BSL name, not the backup.